### DISCUSSION TITLE
Viable approach to handle 301 redirections with static optimization

### DISCUSSION THREAD

**Discussion by alexander-r-jaimy** [vercel/next.js#12193] [Help]
Good afternoon, thanks to zeit team for bringing such a wonderful framework.
I'd like to understand how my current problem can be solved using next.js.

As of now, my project runs on next.js with the usage of custom express server, for a site that meant to be static.
After the release of v9 with awesome `getStaticProps` feature, we are starting to consider to update a project for using SSG and deploy it into AWS CDN instead of paying for s2 instance.

The only reason why we are running a node server behind is 301 redirects that SEO team demands. So it looks like frontend redirects are not an option.

My question is:
Is there any viable solution that can achieve 301 redirects, but still allowing us to convert a project into fully static SSG and serving it from a CDN?

Thank you for your advice.

**Comment by alexander-r-jaimy** [comment]
![Screenshot from 2020-04-26 22-28-52](https://user-images.githubusercontent.com/60174868/80356952-7e9fa200-887a-11ea-8bdf-fde18a9afa27.png)
Some custom example:
we are generating these routes.
We would like to be able to deploy our pages to CDN but still have an ability to make redirects like:
```
/fr/plumbers -> /fr/plombiers
/nl/plumbers -> /nl/loodgieters
```

**Reply by alejandronanez** [reply]
👋 @alexander-r-jaimy, I'm curious, how did you get around this?

**Comment by timneutkens** [comment]
https://github.com/vercel/next.js/discussions/9081 this should help
